src/lib/streams.d.ts

```diff
@@ -0,0 +1,131 @@
+/// Definitions by: Kagami Sascha Rosylight <https://github.com/saschanaz>
+
+interface ReadableStreamSource {
+    start?(controller: ReadableStreamDefaultController): void | Promise<void>;
+    pull?(controller: ReadableStreamDefaultController): void | Promise<void>;
+    cancel?(reason: string): void | Promise<void>;
+}
+
+interface ReadableByteStreamSource {
+    start?(controller: ReadableByteStreamController): void | Promise<void>;
+    pull?(controller: ReadableByteStreamController): void | Promise<void>;
+    cancel?(reason: string): void | Promise<void>;
+
+    type: "bytes";
+}
+
+interface QueuingStrategy {
+    highWaterMark?: number;
+    size?(chunk: ArrayBufferView): number;
+}
+
+declare class ReadableStream {
+    constructor(underlyingSource?: ReadableStreamSource, strategy?: QueuingStrategy);
+    constructor(underlyingSource?: ReadableByteStreamSource, strategy?: QueuingStrategy);
+
+    locked: boolean;
+
+    cancel(reason: string): Promise<void>;
+    getReader(): ReadableStreamDefaultReader;
+    getReader({ mode }: { mode: "byob" }): ReadableStreamBYOBReader;
+    pipeThrough<T extends ReadableStream>({ writable, readable }: { writable: WritableStream, readable: T }): T;

```

Comment:
seem from the [docs](https://streams.spec.whatwg.org/#rs-class-definition), this should take an additional `options` argument.